unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,198 @@
 
-        /// <summary>
-        /// If the dirty flag is true, this coroutine will re-populate all the tracked objects.
-        /// </summary>
+        /// <summary> Adds a rendereable to be tracked and calculating in the culling process. </summary>
+        public void OnRendereableAdded(object sender, Rendereable rendereable) => rendereables.Add(rendereable);
+
+        /// <summary> Sets a layer mask to be ignored, only one layer mask can be ingored at a given time. </summary>
+        public void SetIgnoredLayersMask(int ignoredLayersMask) { this.ignoredLayersMask = ignoredLayersMask; }
+
+        /// <summary> This will re-populate all the tracked objects synchronously </summary>
+        public void ForcePopulateRenderersList()
+        {
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
+
+            ForceCalculateRenderers();
+            ForceCalculateSkinnedRenderers();
+            usingAnimations = Object.FindObjectsOfType<Animation>();
+
+            dirty = false;
+        }
+
+        /// <summary> If the dirty flag is true, this coroutine will re-populate all the tracked objects. </summary>
         public IEnumerator PopulateRenderersList()
         {
             if (!dirty)
                 yield break;
 
-            List<Renderer> renderersList = new List<Renderer>();
-            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
-            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>();
-
-            // TODO new task, avoid FindGameObjectsOfType and use:
-            // SceneManager.GetActiveScene().GetRootGameObjects(), calculate from there.
+            detectedSkinnedRenderers = Object.FindObjectsOfType<SkinnedMeshRenderer>();
 
+            yield return null;
+            yield return CalculateRenderers();
+            yield return null;
+            yield return CalculateSkinnedRenderers();
+            yield return null;
+            usingAnimations = Object.FindObjectsOfType<Animation>();
             yield return null;
 
-            int amount = 0;
-            foreach (Renderer renderer in allRenderers)
+            dirty = false;
+        }
+
+        public void Dispose()
+        {
+            dirty = true;
+            usingRenderers = null;
+            usingSkinnedRenderers = null;
+            detectedSkinnedRenderers = null;
+            usingAnimations = null;
+            PoolManager.i.OnGet -= MarkDirty;
+            Rendereable.OnRendereableAdded -= OnRendereableAdded;
+        }
+
+        /// <summary> Sets the dirty flag to true to make PopulateRenderersList retrieve all the scene objects on its next call. </summary>
+        public void MarkDirty() => dirty = true;
+
+        /// <summary> Returns true if dirty. </summary>
+        public bool IsDirty() => dirty;
+
+        /// <summary> Returns the renderers list. </summary>
+        public ICollection<Renderer> GetRenderers() => usingRenderers;
```

Comment:
Considering how you consume API, collections should be exposed as `IReadOnlyList` (declaring that modifications are not supported)

Reply:
Good point, I will be changing it